lib/ansible/modules/cloud/ovh/ovh_ip_failover.py

```diff
@@ -0,0 +1,197 @@
+    while client.get('/ip/{0}/task'.format(urllib.quote_plus(name)),
+                     function='genericMoveFloatingIp',
+                     status='todo'):
+        time.sleep(1)  # Delay for 1 sec
+        currentTimeout -= 1
+        if currentTimeout < 0:
+            return False
+    return True
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True),
+            service=dict(required=True),
+            endpoint=dict(required=True),
+            application_key=dict(required=True, no_log=True),
+            application_secret=dict(required=True, no_log=True),
+            consumer_key=dict(required=True, no_log=True),
+            timeout=dict(default=120, type='int')
+        ),
+        supports_check_mode=True
+    )
+
+    result = dict(
+        changed=False
+    )
+
+    if not HAS_OVH:
+        module.fail_json(msg='ovh-api python module is required to run this module ')
+
+    # Get parameters
+    name = module.params.get('name')
+    service = module.params.get('service')
+    timeout = module.params.get('timeout')
+
+    # Connect to OVH API
+    client = getOvhClient(module)
+
+    # Check that the load balancing exists
+    try:
+        ips = client.get('/ip', ip=name, type='failover')
+    except APIError as apiError:
+        module.fail_json(
+            msg='Unable to call OVH api for getting the list of ips, '
+                'check application key, secret, consumerkey and parameters. '
+                'Error returned by OVH api was : {0}'.format(apiError))
+
+    if name not in ips and '{0}/32'.format(name) not in ips:
+        module.fail_json(msg='IP {0} does not exist'.format(name))
+
+    # Check that no task is pending before going on
+    try:
+        if not waitForNoTask(client, name, timeout):
+            module.fail_json(
+                msg='Timeout of {0} seconds while waiting for no pending '
+                    'tasks before executing the module '.format(timeout))
+    except APIError as apiError:
+        module.fail_json(
+            msg='Unable to call OVH api for getting the list of pending tasks '
+                'of the ip, check application key, secret, consumerkey '
+                'and parameters. Error returned by OVH api was : {0}'
+                .format(apiError))
+
+    try:
+        ipproperties = client.get('/ip/{0}'.format(urllib.quote_plus(name)))
+    except APIError as apiError:
+        module.fail_json(
+            msg='Unable to call OVH api for getting the properties '
+                'of the ip, check application key, secret, consumerkey '
+                'and parameters. Error returned by OVH api was : {0}'
+            .format(apiError))
+
+    if ipproperties['routedTo']['serviceName'] != service:
+        if not module.check_mode:
+            client.post('/ip/{0}/move'.format(urllib.quote_plus(name)), to=service)
+            time.sleep(3)  # Delay for 3 secs to wait for task created
```

Comment:
The 3 sec delay is not good.

We have to check with an api call if the task is created. if not, continue polling until timeout.

I just tested the module, I get a changed=true, whereas the ip was not moved yet. What happened:
- the module created the task
- the module wait for 3 sec (but the task is still not created yet after 3 sec)
- the module wait for no task (waitForNoTask function)
- the module immediately returns true for waitForNoTask, because there is no task with status='todo' and function='genericMoveFloatingIp' (not created yet after 3 sec)
- the module ends with changed=true, but the ip move is not finished yet on the OVH side

Reply:
Ok i get it. When the task is marked completed it does not mean the move has been done. There is a delay before it's really done by OVH backend. The move can be more or less long depending of the backend's workload.

What we usually do in production after FO move : we use a bash script testing that the move is done (by calling http request for instance)

Reply:
https://api.ovh.com/console/#/ip/%7Bip%7D/task#GET

status may be in:
- cancelled
- customerError
- doing
- **done**
- init
- ovhError
- todo

When a task has the status _done_, the task is completed on the ovh backend. I use this when I write deployment scripts and the task status from the api reflects the real ip state. This should be handled by this module : the module exit only when we are 100% sure the ip has moved to its new location.

I am sorry I insist, and thanks for your work

Reply:
oh good to know, I'll try it  !

Reply:
Your comment bumped a feature I wanted to code. The idea would be to move a set of IPs without waiting for completion, then waiting for completion of this set in a second time.

The idea would be add a parameter "wait_completion". If false, the module will not wait for the task to be completed, if true it will wait the "done" status. In this case, the module will return the created taskId

Another parameter "wait_task_completion" allows to pass a previously created taskId to wait for its completion.

By combining the two parameters, we manage to move a set of IPs then wait for completion if more optimized manner than doing it sequentially.

I'll commit code as soon as it works fine.